client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoDriverUri.java

```diff
@@ -357,6 +362,18 @@ public void setupClient(OkHttpClient.Builder builder)
         }
     }
 
+    private static DnsResolver instantiateDnsResolver(Class<? extends DnsResolver> resolverClass, Optional<String> context)
+    {
+        try {
+            return context.isPresent() ?
+                resolverClass.getConstructor(String.class).newInstance(context.get()) :
+                resolverClass.getDeclaredConstructor().newInstance();
+        }
+        catch (Exception e) {
+            throw new RuntimeException("Unable to instantiate custom DNS resolver " + resolverClass.getName(), e);
```

Comment:
Throw `ClientException` here so that the original message is returned to the user